packages/core/strapi/lib/types/__tests__/definitions/utils/expression.d.ts

```diff
@@ -0,0 +1,168 @@
+import type { Utils } from '@strapi/strapi';
+import { Array } from '../../../../services/entity-service/types/params/filters/operators';
+
+// IsNever
+type IsNeverGivenNever = Utils.Expression.IsNever<never>;
+type IsNeverNotGivenNever = Utils.Expression.IsNever<string>;
+type IsNeverGivenUnknown = Utils.Expression.IsNever<unknown>;
+
+// IsNotNever
+type IsNotNeverGivenNever = Utils.Expression.IsNotNever<never>;
+type IsNotNeverGivenUnknown = Utils.Expression.IsNotNever<unknown>;
+type IsNotNeverGivenString = Utils.Expression.IsNotNever<string>;
+type IsNotNeverGivenStringLiteral = Utils.Expression.IsNotNever<'test'>;
+
+// IsTrue
+type IsTrueGivenTrue = Utils.Expression.IsTrue<true>;
+type IsTrueGivenFalse = Utils.Expression.IsTrue<false>;
+type IsTrueGivenBoolean = Utils.Expression.IsTrue<boolean>;
+type IsTrueGivenNonBoolean = Utils.Expression.IsTrue<number>;
+type IsTrueGivenNonBooleanLiteral = Utils.Expression.IsTrue<10>;
+type IsTrueGivenOne = Utils.Expression.IsTrue<1>;
+type IsTrueGivenZero = Utils.Expression.IsTrue<0>;
+
+// IsFalse
+type IsFalseGivenTrue = Utils.Expression.IsFalse<true>;
+type IsFalseGivenFalse = Utils.Expression.IsFalse<false>;
+type IsFalseGivenBoolean = Utils.Expression.IsFalse<boolean>;
+type IsFalseGivenNonBoolean = Utils.Expression.IsFalse<number>;
+type IsFalseGivenNonBooleanLiteral = Utils.Expression.IsFalse<10>;
+type IsFalseGivenOne = Utils.Expression.IsFalse<1>;
+type IsFalseGivenZero = Utils.Expression.IsFalse<0>;
+
+// Strict Equal
+type IsStrictEqualGivenNever = Utils.Expression.StrictEqual<never, never>;
+type IsStrictEqualGivenNeverAndUnknown = Utils.Expression.StrictEqual<never, unknown>;
+type IsStrictEqualGivenEqualLiterals = Utils.Expression.StrictEqual<1, 1>;
+type IsStrictEqualGivenEqualTypes = Utils.Expression.StrictEqual<boolean, boolean>;
+type IsStrictEqualGivenDifferentLiterals = Utils.Expression.StrictEqual<1, 2>;
+type IsStrictEqualGivenDifferentTypes = Utils.Expression.StrictEqual<boolean, string>;
+
```

Comment:
Could we add a test for StrictEqual<'hello', string> or StrictEqual<1, number> to see if it ignores extendable types?